resolwe_bio/processes/chip_seq/macs2.yml

```diff
@@ -445,31 +485,258 @@
+            {% endif %}
+        {% else %}
+            re-save-file case_prepeak_qc "${NAME}_prepeak_qc_report.txt"
+            TAGALIGN_OUT_FIELD="case_tagalign"
+            {% if prepeakqc_settings.tn5 %}
+              re-save-file ${TAGALIGN_OUT_FIELD} "${NAME}_tn5.tagAlign.gz"
+              CASE_TAGALIGN="${NAME}_tn5.tagAlign.gz"
+            {% else %}
+              re-save-file ${TAGALIGN_OUT_FIELD} "${NAME}.tagAlign.gz"
+              CASE_TAGALIGN="${NAME}.tagAlign.gz"
+            {% endif %}
+        {% endif %}
+
+      {% endfor %}
+
       BROAD_CUTOFF='{% if settings.broad_cutoff %}--broad-cutoff {{settings.broad_cutoff}}{% endif %}'
 
       macs2 callpeak \
-        -t {{ case_prepeak.tagalign.file or case.bam.file }} \
-        {% if control or control_prepeak %}-c {{ control_prepeak.tagalign.file or control.bam.file }}{% endif %} \
-        -f {{ "BED" if case_prepeak else "BAM" }} \
-        -n "${NAME}" \
+        -t {% if tagalign %} "${CASE_TAGALIGN}" {% else %} {{ case.bam.file }} {% endif %} \
+        {% if control %}-c {% if tagalign %} "${CONTROL_TAGALIGN}" {% else %} {{ control.bam.file }} {% endif %}{% endif %} \
+        -f {{ "BED" if tagalign else "BAM" }} \
+        -n "${CASE_NAME}" \
         --gsize {{ gsize }} \
-        {% if settings.duplicates or case_prepeak %}--keep-dup {{ settings.duplicates or settings.duplicates_prepeak }}{% endif %} \
-        {% if settings.qvalue %}-q {{ settings.qvalue }}{% endif %} \
-        {% if settings.pvalue or settings.pvalue_prepeak and not setting.qvalue %}-p {{ settings.pvalue or settings.pvalue_prepeak }}{% endif %} \
+        {% if settings.duplicates or tagalign %}--keep-dup {{ settings.duplicates or settings.duplicates_prepeak }}{% endif %} \
+        {% if settings.qvalue and not tagalign %}-q {{ settings.qvalue }}{% endif %} \
+        {% if settings.pvalue and not setting.qvalue %}-p {{ settings.pvalue }} {% elif tagalign and not setting.qvalue %} -p {{settings.pvalue_prepeak}} {% endif %} \
         {% if settings.mfold_lower %}-m {{ settings.mfold_lower }} {{ settings.mfold_upper }}{% endif %} \
         {% if settings.nolambda %}--nolambda{% endif %} \
         {% if settings.slocal %}--slocal {{ settings.slocal }}{% endif %} \
         {% if settings.llocal %}--llocal {{ settings.llocal }}{% endif %} \
         {% if settings.fix_bimodal %}--fix-bimodal{% endif %} \
-        {% if settings.nomodel or (settings.nomodel_prepeak and case_prepeak) %}--nomodel{% endif %} \
-        {% if settings.extsize or case_prepeak %}--extsize {{ settings.extsize or case_prepeak.fraglen }}{% endif %} \
+        {% if settings.nomodel or (settings.nomodel_prepeak and tagalign) %}--nomodel{% endif %} \
+        {% if settings.extsize or tagalign %}--extsize {% if settings.extsize %} {{ settings.extsize }} {% else %} ${FRAGLEN} {% endif %}{% endif %} \
         {% if settings.shift %}--shift {{ settings.shift }}{% endif %} \
```

Comment:
Maybe add `is defined` to ifs. I can see a case where `extsize` or `fraglen` might be 0.

Reply:
Solved here and for other `IFs` a bit differently to account for the use with the workflows.